### ISSUE TITLE
Extra features/enhancement for dual list selector

### ISSUE THREAD

**Issue by MariSvirik** [patternfly/patternfly-design#1033]
I have a use case where we need some additional features in the Dual list selector component. The use case is adding/removing Ansible roles in the Host detail page in Satellite.

1. As there can be more than 100 items on one side of the dual list, it's preferred to use **pagination components.** Combination of an endless scroll and search bar without knowing the name of the item is not a user-friendly option.
2. The order of the added items on the right side of the dual list is essential. That's the reason I propose the '**drag and drop feature**' and **ordered list** to be added to the component. Without this feature, users would be forced to add the items in the correct order right from the beginning and face an arduous task fixing the item order if they make a mistake.

**My proposed design**

![DUal list](https://user-images.githubusercontent.com/51960547/115363152-83174600-a1c2-11eb-814b-bfb195d25fb8.png)

**Comment by mcarrano** [comment]
Seems like a reasonable request @MariSvirik . I will need to get it on the roadmap. Is this needed in a specific timeframe? Right now I am scheduling out into Q3.

**Comment by MariSvirik** [comment]
@mcarrano I was told we need to build the entire Ansible part in 5 months. Adding and removing Ansible roles (that is done via Dual list) is a crucial part of that implementation. Therefore if it's possible to prioritize at least an 'ordered list' and the 'drag and drop' features and schedule it for the very beginning of Q3, that would be great. The pagination could be added later.

**Comment by MariSvirik** [comment]
@mcarrano I updated the issue according to the feedback during design share.

**Comment by mcarrano** [comment]
@MariSvirik I reviewed this with our React devs. We should be able to support reordering items in the right-hand list as in the design. Changing this to an ordered (numbered) list will be a problem, so I don't expect we can support that. IMO, the numbers are not needed as long as the user can reorder the items in the list by dragging.

We have had some problems with our existing drag & drop implementation and we are currently working on integrating a 3rd party library to improve our interaction. Once that work is complete, we can get this enhancement into a release. I will expedite it, but I'm thinking that the start of Q4 is a good bet.

**Comment by xprazak2** [comment]
Thank you for doing this. The numbers are nice to have but not a necessity.

**Comment by mcarrano** [comment]
The result would look something like this:

![Dual-list-reorderable.png](https://images.zenhubusercontent.com/5ce816524e6c594dc1b0df4d/5f8927b4-bac0-47a4-843a-9c1e6e19a048)

If we are all good with this, I will open a core issue to start implementation.

cc @mcoker

**Comment by mcarrano** [comment]
Opened https://github.com/patternfly/patternfly/issues/4319 to begin core implementation.
